### ISSUE TITLE
is.lengthyArray

### ISSUE THREAD

**Issue by rarkins** [sindresorhus/is#53]
Would you support a PR for is.lengthyArray? Ie returns true if is.array(input) && input.length

**Comment by sindresorhus** [comment]
This already exists as `!is.empty(array)`.

**Comment by rarkins** [comment]
Thanks, that looks good

**Comment by rarkins** [comment]
Actually, I don't think it's the same. For example, `!is.empty('b')` returns `true` while it is not an array.

**Comment by brandon93s** [comment]
Using `is.array(x) && !is.empty(x)` reads nicely and I believe accomplishes the goal. If you're using it often, a local helper could support the use case:

```js
const isNonEmptyArray = x => is.array(x) && !is.empty(x) 
```

**Comment by rarkins** [comment]
I was hoping that `is` would remove the need for commonly-used local helper functions and checking that `x` is a non-empty array is a fairly common use case. 

If the criteria for rejecting proposals is that they're composed of two other ones, then `.nullOrUndefined(value)` reads even nicer:

```js
const isNullOrUndefined => x => is.null(x) || is.undefined(x);
```

Another example (I think):
```js
const isEmptyObject => x => is.object(x) && is.empty(x);
```

**Comment by sindresorhus** [comment]
I'm starting to think `is.empty()` was a mistake. I can't really see its use-case without first checking the type.

I'm leaning towards creating individual `is.emptyArray`/`is.nonEmptyArray` like methods for String, Array, Object, Map, Set.

Thoughts?

/// @SamVerschueren

**Comment by SamVerschueren** [comment]
I agree that `is.empty()` doesn't really say what it checks. With `Ow`, we have the benefit that you have to specify the type first, so `ow.string.empty()` really makes sense. But with `is`, we don't have that so I think we should make it `ow.emptyString()`, `ow.emptyArray()`, ...

**Comment by sindresorhus** [comment]
Alright. PR welcome to remove `is.empty` and add the mentioned methods.

**Comment by ArfatSalman** [comment]
I am taking this up.
